principles.rst

```diff
@@ -36,55 +36,273 @@ but lacks the unarticulated principles, which will be added over time.
 All principles in this document are linked back to the PR that introduce them,
 so that readers can learn the context in which they were written.
 
+We use a notational convention here where each principle is given an abbreviation,
+but we use its full spelling wherever possible. The full spelling is useful for readers,
+given the multitude of principles in play. Including the abbreviation is useful in case
+the abbreviation is used in other fora (e.g. email, GitHub, etc.) to allow the curious
+to find (e.g. with Ctrl+F) the abbreviation within this page.
+
 Accepted principles
 -------------------
 
-.. _`#378`: https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0378-dependent-type-design.rst
+.. _`#281`: proposals/0281-visible-forall.rst
+.. _`#378`: proposals/0378-dependent-type-design.rst
+.. _`#448`: proposals/0448-type-variable-scoping.rst
 
 Syntax
 ~~~~~~
 
-.. _SUP:
+Syntactic Unification Principle (SUP)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. _`Syntactic Unification Principle`:
 
-**Syntactic Unification Principle (SUP).** (From `#378`_.) In the absence of punning, there is
+**Principle:** In the absence of punning, there is
 no difference between type-syntax and term-syntax.
 
-The SUP_ is a *long term* goal; today, there are many, many violations of this principle. (For example,
+The `Syntactic Unification Principle`_ is a *long term* goal; today, there are many, many violations of this principle. (For example,
 ``case`` can never appear in types, and ``Int -> Bool`` is meaningless in a term.) However, this principle
 should direct future proposals not to make the situation worse, and proposals that bring us closer to
-the SUP_ are to be valued.
+the `Syntactic Unification Principle`_ are to be valued.
 
-Name resolution
-~~~~~~~~~~~~~~~
+*Motivation:* The `Syntactic Unification Principle`_ keeps us forward-compatible with a possible future where the
+distinction between term-syntax and type-syntax is removed.
+
+From `#378`_.
+
+Explicit Variable Principle (EVP)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. _`Explicit Variable Principle`:
+
+**Principle:** It is possible to write out all (specified)
+type arguments in every polymorphic function application,
+give the type for every bound variable,
+and write a type signature for every expression. This requires the ability to
+bring type variables into scope. These variables can be brought into scope
+without relying on matching or unification.
+
+Examples::
+
+  const :: a -> b -> a
+  const x y = ...    -- there must be some way to name the types of x and y here
+  -- using `const (x :: a) (y :: b) = ...` is not powerful enough, because it relies
```

Comment:
In these examples it would help to show what it would take to bring the examples in line with the principle, e.g. for `const` would it suffice if we could write the following?

```
const :: a -> b -> a
const @a @b (x :: a) (y :: b) = x
```